src/1_api.js

```diff
@@ -71,6 +71,22 @@ Server.prototype.getUrl = function(resource, data) {
 			}
 		}
 	}
+	// check for branch_key then app_id here
+	var branch_id = /^[0-9]{15,20}$/;
+	var branch_key = /((?:[a-z][a-z]*[0-9]+[a-z0-9]*))/;
```

Comment:
we can use a stricter regex here: `/key_(live|test)_[A-Za-z0-9]{32}/` is the actual pattern